client/src/js/components/bhCurrencySelect.js

```diff
@@ -0,0 +1,98 @@
+angular.module('bhima.components')
+.component('bhCurrencySelect', {
+  controller : bhCurrencySelect,
+  templateUrl : 'partials/templates/bhCurrencySelect.tmpl.html',
+  bindings : {
+    validationTrigger : '<',
+    currencyId : '=',
+    disableIds : '<',
+    onChange : '&'
+  }
+});
+
+bhCurrencySelect.$inject = [ '$scope', 'CurrencyService' ];
```

Comment:
I use $scope.$watch().  Can you give some some example code that does the same thing and doesn't use $scope.$watch()?